src/array/utf8/mutable.rs

```diff
@@ -481,33 +526,111 @@ where
     O: Offset,
     P: AsRef<str>,
     I: Iterator<Item = P>,
+{
+    let mut offsets = MutableBuffer::<O>::with_capacity(1);
```

Comment:
We can estimate this capacity to be `1 + iterator.size_hint().unwrap()`, which avoids an extra allocation :)

Reply:
This is neat, made the change, thanks